### PR TITLE
fix: Single integration test for deploy campaign

### PR DESCRIPTION
**Pull request by JamesEarle**
Single integration test for deploy campaign that runs deployment and then also proceeds to run through several crucial flows as part of verification.

[For 339](https://wilderworld.atlassian.net/browse/MUD-339)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "@nomicfoundation/hardhat-network-helpers": "^1.0.8",
     "@nomicfoundation/hardhat-toolbox": "2.0.2",
     "@nomiclabs/hardhat-ethers": "2.2.2",
-    "@nomiclabs/hardhat-etherscan": "3.0.0",
+    "@nomiclabs/hardhat-etherscan": "3.1.7",
```
Comment (Whytecrowe) [inline package.json:47]: you don't get an error with this? I thought this package is already part of the `hardhat-toolbox`, no?

File: test/DeployCampaign.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    config.mockMeowToken = hre.network.name === "hardhat";
+    const campaign = await runZnsCampaign({ config, dbVersion: "1.0.0" });
+
+    zns = campaign.state.contracts;
+
+    //  CurvePricer, stake, open
+    distConfig = {
+      pricerContract: zns.curvePricer.address,
+      paymentType: 1,
+      accessType: 1,
```
Comment (Whytecrowe) [inline test/DeployCampaign.integration.test.ts]: please use existing enums already created for each of these fields:

https://github.com/zer0-os/zNS/blob/feat/verify-monitor-contracts/test/helpers/constants.ts#L20-L37

File: test/DeployCampaign.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+      users,
+      domains,
+      domainAddressMock.address,
+      "https://zns.domains/", // tokenUri
+      distConfig,
+      config.rootPriceConfig,
+      zns
+    );
+
+    const balanceAfterPromises = [
```
Comment (Whytecrowe) [inline test/DeployCampaign.integration.test.ts:150]: it can be the same array as before.

File: test/helpers/deploy-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  domains : Array<string>,
+  domainAddress : string,
+  tokenUri : string,
+  distConfig : IDistributionConfig,
+  priceConfig : ICurvePriceConfig,
+  zns : TZNSContractState,
+) : Promise<void> => {
+  let index = 0;
+
+  // TODO need to return state for outer calling tests, better way to do this?
```
Comment (Whytecrowe) [inline test/helpers/deploy-helpers.ts]: what is this for? still needs to be done?

File: package.json (hunk 1 above)
Comment (JamesEarle) [reply package.json:47]: I had an error without adding this, and this was the resolution. I couldn't setup a Sepolia network anymore in the HH config without upgrading the package

File: test/helpers/deploy-helpers.ts (hunk 4 above)
Comment (JamesEarle) [reply test/helpers/deploy-helpers.ts]: No I resolved that it was more meant to be explanatory. I couldn't get the state of the contracts in the scope of the code that calls `registerRootDomainBulk` to reflect the changes made within that function if I used a prototype function like `.map()` or `.reduce()`, I'm sure there is a way but it wouldn't work for me and I went with this instead. Removed the comment

File: test/DeployCampaign.integration.test.ts (hunk 3 above)
Comment (JamesEarle) [reply test/DeployCampaign.integration.test.ts:150]: I don't think this is true. I tried this to confirm and it only returns the before values when we don't have a second "after" array even if they are all the same promises.

When using a single promise array it fails the test with this error, and when using two different arrays (containing the same promises) it passes the tests. Must be another JavaScript behind the scenes thing, trying to save time and can see that the promises in the array are already resolved and so even with `await Promise.all()` running a second time, it just returns the resolved values
 
image

**Pull request merged**